alf/algorithms/off_policy_algorithm.py

```diff
@@ -105,18 +101,41 @@ def _train_iter_off_policy(self):
         """User may override this for their own training procedure."""
         config: TrainerConfig = self._config
 
+        first_iter = (alf.summary.get_global_counter() == 0)
+
         if not config.update_counter_every_mini_batch:
             alf.summary.get_global_counter().add_(1)
 
-        with torch.no_grad():
-            with record_time("time/unroll"):
-                while True:
-                    training_info = self.unroll(config.unroll_length)
-                    self.summarize_rollout(training_info)
-                    self.summarize_metrics()
-                    if self._exp_replayer.total_size >= config.initial_collect_steps:
-                        break
+        with record_time("time/unroll"):
+            while True:
+                training_info = self.unroll(config.unroll_length)
+                self.summarize_rollout(training_info)
+                self.summarize_metrics()
+                if self._exp_replayer.total_size >= config.initial_collect_steps:
+                    break
+
+        steps = self.train_from_replay_buffer(True)
+
+        with record_time("time/after_train_iter"):
+            # skip the first iteration as it can unroll quite long
```

Comment:
I don't understand the reason to skip.

Reply:
Because we have init_collect steps. It might be thousands of steps. I think I need to fix this place to make sure that init_collect_steps is also chopped into multiples of unroll_length. Otherwise the memory will explode.